src/Analyzers/CSharp/CodeFixes/ConvertNamespace/ConvertNamespaceTransform.cs

```diff
@@ -243,28 +337,68 @@ private static FileScopedNamespaceDeclarationSyntax ConvertNamespaceDeclaration(
-                fileScopedNamespace.NamespaceKeyword,
-                fileScopedNamespace.Name,
-                SyntaxFactory.Token(SyntaxKind.OpenBraceToken).WithTrailingTrivia(fileScopedNamespace.SemicolonToken.TrailingTrivia),
-                fileScopedNamespace.Externs,
-                fileScopedNamespace.Usings,
-                fileScopedNamespace.Members,
-                SyntaxFactory.Token(SyntaxKind.CloseBraceToken),
-                semicolonToken: default).WithAdditionalAnnotations(Formatter.Annotation);
-
-            // Ensure there is no errant blank line between the open curly and the first body element.
-            var firstBodyToken = namespaceDeclaration.OpenBraceToken.GetNextToken();
-            if (firstBodyToken != namespaceDeclaration.CloseBraceToken &&
-                firstBodyToken.Kind() != SyntaxKind.EndOfFileToken &&
-                HasLeadingBlankLine(firstBodyToken, out var firstBodyTokenWithoutBlankLine))
+            var nameSyntax = fileScopedNamespace.Name.WithAppendedTrailingTrivia(fileScopedNamespace.SemicolonToken.LeadingTrivia);
+            SyntaxToken openBraceToken;
+            if (newLinePlacement.HasFlag(NewLinePlacement.BeforeOpenBraceInTypes))
             {
-                namespaceDeclaration = namespaceDeclaration.ReplaceToken(firstBodyToken, firstBodyTokenWithoutBlankLine);
+                nameSyntax = nameSyntax.WithAppendedTrailingTrivia(SyntaxFactory.EndOfLine(lineEnding));
+                openBraceToken = SyntaxFactory.Token(SyntaxKind.OpenBraceToken).WithoutLeadingTrivia().WithTrailingTrivia(fileScopedNamespace.SemicolonToken.TrailingTrivia);
             }
+            else
+            {
+                nameSyntax = nameSyntax.WithAppendedTrailingTrivia(SyntaxFactory.Space);
+                openBraceToken = SyntaxFactory.Token(SyntaxKind.OpenBraceToken).WithoutLeadingTrivia().WithTrailingTrivia(fileScopedNamespace.SemicolonToken.TrailingTrivia);
+            }
+
+            if (openBraceToken.TrailingTrivia is not [.., { RawKind: (int)SyntaxKind.EndOfLineTrivia }])
+            {
+                openBraceToken = openBraceToken.WithAppendedTrailingTrivia(SyntaxFactory.EndOfLine(lineEnding));
+            }
+
+            FileScopedNamespaceDeclarationSyntax adjustedFileScopedNamespace;
+            var closeBraceToken = SyntaxFactory.Token(SyntaxKind.CloseBraceToken).WithoutLeadingTrivia().WithoutTrailingTrivia();
+
+            // Normally the block scoped namespace will have a newline after the closing brace. The only exception to
+            // this occurs when there are no tokens after the closing brace and the document with a file scoped
+            // namespace did not end in a trailing newline. For this case, we want the converted block scope namespace
+            // to also terminate without a final newline.
+            if (!fileScopedNamespace.GetLastToken().GetNextTokenOrEndOfFile().IsKind(SyntaxKind.EndOfFileToken)
```

Comment:
i'm struggling to thnik of any case where a file-scoped namespace is not followed by EOF.  The idea of them is that they parse all the way to the end of the file.  so i think you can always just grab the EOF and use it (which i think would clear things up too).

Reply:
➡️ Answered in https://github.com/dotnet/roslyn/pull/68428/files#r1227032919